src/server.ts

```diff
@@ -1,226 +1,385 @@
 import {
-	createConnection,
-	TextDocuments,
-	Diagnostic,
-	DiagnosticSeverity,
-	ProposedFeatures,
-	InitializeParams,
-	DidChangeConfigurationNotification,
-	CompletionItem,
-	CompletionItemKind,
-	TextDocumentPositionParams,
-	TextDocumentSyncKind,
-	InitializeResult
+    createConnection,
+    TextDocuments,
+    Diagnostic,
+    DiagnosticSeverity,
+    ProposedFeatures,
+    InitializeParams,
+    DidChangeConfigurationNotification,
+    CompletionItem,
+    CompletionItemKind,
+    TextDocumentPositionParams,
+    TextDocumentSyncKind,
+    InitializeResult,
+    Location,
+    Hover,
+    SignatureHelp,
+    SignatureInformation,
+    ParameterInformation,
+    DocumentFormattingRequest,
+    DocumentSymbol,
+    SymbolKind,
+    DocumentSymbolParams,
+    FormattingOptions,
+    TextEdit,
+    Position
 } from 'vscode-languageserver/node';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import * as cp from 'child_process';
+import * as path from 'path';
+import * as os from 'os';
+import * as fs from 'fs';
 
-import {
-	TextDocument
-} from 'vscode-languageserver-textdocument';
-
-// Create a connection for the server, using Node's IPC as a transport.
-// Also include all preview / proposed LSP features.
+// Language Server Connection
 const connection = createConnection(ProposedFeatures.all);
-
-// Create a simple text document manager.
 const documents: TextDocuments<TextDocument> = new TextDocuments(TextDocument);
 
-let hasConfigurationCapability = false;
-let hasWorkspaceFolderCapability = false;
-let hasDiagnosticRelatedInformationCapability = false;
+// Configuration Types
+interface ParasailSettings {
+    maxNumberOfProblems: number;
+    enableFormatting: boolean;
+    libraryPaths: string[];
+    implicitImports: boolean;
+}
+
+// Language Constants
+const PARASAIL_KEYWORDS: { [key: string]: string } = {
+    "func": "Defines a function: `func name(params) -> return_type is ... end func`",
+    "type": "Defines a type: `type Name is ... end type`",
+    "interface": "Declares an interface: `interface Name is ... end interface`",
+    "class": "Defines a class: `class Name is ... end class`",
+    "operator": "Operator overload: `operator \"=\"(Left: Type, Right: Type) -> Boolean is ...`",
+    "package": "Module declaration: `package Name is ... end package`",
+    "const": "Constant declaration: `const Name: Type := Value`",
+    "var": "Variable declaration: `var Name: Type := Value`",
+    "abstract": "Abstract operation: `abstract func Name(...)`",
+    "extends": "Inheritance: `class Name extends Parent`",
+    "exports": "Visibility control: `exports {Name1, Name2}`",
```

Comment:
Exports is a keyword that is used in a class to determine which functions are public. These functions correspond to the functions defined in the interface

```
interface A<> is
   func Test();
end interface A

...

class A is
   const My_Private_Var := "Private";

   func Private_Func() is
      ...
   end func Private_Func
exports
   func Test() is
      ...
   end func Test
end class A
```